content/en/platform/corda/5.0/deploying-operating/observability/metrics.md

```diff
@@ -49,38 +51,166 @@ metrics:
+| :----------- | :----------- | :----------- | :----------- |
+| `corda_messaging_processor_time_seconds` | Timer | <ul><li>`messagepattern.type`</li><li>`messagepattern.clientid`</li><li>`operation.name`</li></ul> | The time spent in the consumer’s `onNext` or `onSnapshot` functions. The following subscription processors have this metric wrapping the calls to `onNext` functions: <ul><li>`PubSubSubscriptionImpl`</li><li>`CordaRPCSenderImpl`</li><li>`CompactedSubscriptionImpl` (`onNext` and `onSnapshot`)</li><li>`EventLogSubscriptionImpl`</li><li>`PubSubSubscriptionImpl`</li><li>`RPCSubscriptionImpl`</li><li>`StateAndEventSubscriptionImpl`</li></ul> |
+| `corda_messaging_batch_size` |  | <ul><li>`messagepattern.type (StateAndEvent)`</li><li>`messagepattern.clientid`</li></ul> | The state and event subscription uses this metric to measure the size of batches polled from `kafka.(StateAndEventSubscriptionImpl)`. |
+| `corda_messaging_poll_time_seconds` | Timer | <ul><li>`messagepattern.type (StateAndEvent)`</li><li>`messagepattern.clientid`</li><li>`operation.name (statePoll)`</li></ul> | State and event consumer state poll time. |
+| `corda_messaging_poll_time_seconds` | Timer | <ul><li>`messagepattern.type (StateAndEvent)`</li><li>`messagepattern.clientid`</li><li>`operation.name (eventPoll)`</li></ul> | State and event consumer event poll time. |
+
+
+#### Flow
+
+| Metric | Type | Tags | Description |
+| :----------- | :----------- | :----------- | :----------- |
+| `corda_flow_run_time_seconds` | Timer | <ul><li>`virtualnode`</li><li>`flow.class`</li><li>`status`</li></ul> | The time it took for a flow to complete successfully or to produce an error. |
+| `corda_flow_fiber_serialization_time_seconds` | Timer | <ul><li>`flow.class`</li></ul> | The time it took to serialize a flow fiber. |
+| `corda_flow_fiber_deserialization_time_seconds` | Timer | <ul><li>`flow.class`</li></ul> | The time it took to serialize a flow fiber. |
+| `corda_flow_start_lag` |  | <ul><li>`flow.class`</li></ul> | The lag between flow start event, the REST API, and the flow processor. |
+| `corda_flow_execution_time_seconds` | Timer | <ul><li>`flow.class`</li><li>`status`</li></ul> | The time it took to execute the flow (excluding any start lag). |
+| `corda_flow_event_lag` |  | <ul><li>`flow.class`</li><li>`flow.event`</li></ul> | The lag between flow event publication and processing. |
+| `corda_flow_event_pipeline_execution_time_seconds` | Timer | <ul><li>`flow.class`</li><li>`flow.event`</li></ul> | The time it took to execute the pipeline for given flows and flow event types. |
+| `corda_flow_event_fiber_execution_time_seconds` | Timer | <ul><li>`flow.class`</li></ul> | The time it took to execute the fiber for a single suspension point. |
+| `corda_flow_pipeline_execution_time_seconds` | Timer | <ul><li>`flow.class`</li></ul> | A total time that a flow spent processing in the pipeline, rather than queued (includes fiber execution time.) |
```

Comment:
```suggestion
| `corda_flow_pipeline_execution_time_seconds` | Timer | <ul><li>`flow.class`</li></ul> | The total time that a flow spent processing in the pipeline, rather than queued (includes fiber execution time.) |
```